docs/design/features/InlineArrayAttribute.md

```diff
@@ -0,0 +1,93 @@
+### Background and motivation
+
+There were multiple proposals and ideas in the past asking for a no-indirection primitive for inline data. [Example1(inline strings)](https://github.com/dotnet/csharplang/issues/2099),  [Example2(inline arrays)](https://github.com/dotnet/runtime/issues/12320), [Example3(generalized fixed-buffers)](https://github.com/dotnet/csharplang/blob/main/proposals/low-level-struct-improvements.md#safe-fixed-size-buffers)
+Our preexisting offering in this area – [unsafe fixed-sized buffers](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/unsafe-code#fixed-size-buffers) has multiple constraints, in particular it works only with blittable value types and provides no overrun/type safety, which considerably limits its use.
+
+*The InlineArrayAttribute is a building block to allow efficient, type-safe, overrun-safe indexable/sliceable inline data.*
```

Comment:
The InlineArrayAttribute alone does not address overrun-safety. 

Related to https://github.com/dotnet/runtime/issues/105586.

Reply:
By itself InlineArray provides only storage. Thus I call it a "building block", not "a type-safe, overrun-safe indexable data". It is not that.

Since the type does not know its length, the indexing and range checking is the responsibility of the higher level implementation (hand-made or compiler-generated) that wraps the storage. 

In a way the C# provided indexing is still safe as one cannot do `arr[5]` on an array of length==3, it is just enforced by the layer that is above IL/typesystem.

This is one tradeoff that we took when we did not go the `InlineArray<T, Length>` path. 
That would have indexing and range checking at the runtime level - without wrappers/helpers.

I think "building block" is vague enough to hint at the purpose of the feature.
We do not need to prescribe here how one can actually build indexable things with this, but including a link to C# feature design/spec would be a good example.

Reply:
>  including a link to C# feature design/spec would be a good example.

Added.